src/main/world/turbo.gd

```diff
@@ -0,0 +1,199 @@
+extends KinematicBody
+"""
+The 3D object representing 'Turbo', the player-controlled character who moves around the overworld.
+"""
+
+# Number from [0.0, 1.0] which determines how quickly Turbo slows down
+const FRICTION := 0.15
+
+# How fast can Turbo move
+const MAX_RUN_SPEED := 6
+
+# How fast can Turbo slip
+const MAX_SLIP_SPEED := 9
+
+# How fast Turbo slips off of slippery platforms
+const MAX_SLIP_ACCELERATION := 220
+
+# How fast can Turbo accelerate
+const MAX_RUN_ACCELERATION := 150
+
+# How slow can Turbo move before she stops
+const MIN_RUN_SPEED := 4
+
+# How fast does gravity accelerate us
+const GRAVITY := 40.0
+
+# Vertical force applied to Turbo when she jumps
+const JUMP_SPEED := 16
+
+var _velocity := Vector3(0, 0, 0)
+
+# 'true' if the controls should be rotated 45 degrees, giving the player orthographic controls.
+# 'false' if the controls should not be rotated, giving the player isometric controls.
+var _rotate_controls := true
+
+# 'true' if Turbo is currently mid-air after having jumped
+var _jumping := false
+
+# 'true' if Turbo is either slipping, or mid-air after having slipped
+var _slipping := false
+
+func _physics_process(delta):
+	var was_on_floor = is_on_floor()
+	
+	_apply_friction()
+	_apply_gravity(delta)
+	_apply_player_input(delta)
+
+	move_and_slide(_velocity, Vector3.UP)
+	
+	_slide_from_narrow_surfaces(delta)
+
+	if is_on_floor():
+		if (Input.is_action_pressed("jump") or not $JumpBuffer.is_stopped()) and not _slipping:
+			jump()
+		else:
+			_jumping = false
+	
+	if is_on_floor() and not _jumping:
+		# the character visibly jitters between y values of [1.06, 1.10] if we do not smooth the value
+		translation.y = stepify(translation.y, 0.2)
+		
+	if not is_on_floor() and was_on_floor and not _jumping and not _slipping:
+		$CoyoteTimer.start()
+
+
+func _apply_gravity(delta: float) -> void:
+	if is_on_floor():
+		_velocity.y = max(_velocity.y, 0)
+	_velocity += Vector3.DOWN * GRAVITY * delta
+
+
+func _apply_friction() -> void:
+	if Input.is_action_pressed("ui_left") \
+		or Input.is_action_pressed("ui_right") \
+		or Input.is_action_pressed("ui_up") \
+		or Input.is_action_pressed("ui_down"):
+		# Turbo is currently running; no friction when they're running
+		pass
+	elif $CoyoteTimer.is_stopped() and not is_on_floor():
+		# Turbo is in mid-air; no friction in mid-air
+		pass
+	elif _slipping:
+		# Turbo is slipping; no friction when slipping
+		pass
+	else:
+		# apply friction
+		var xy_velocity: Vector2 = lerp(_get_xy_velocity(), Vector2(0, 0), FRICTION)
+		if xy_velocity.length() < MIN_RUN_SPEED:
+			xy_velocity = Vector2()
+		_set_xy_velocity(xy_velocity)
+
+
+"""
+Translates Turbo's (X, Y, Z) velocity where 'Y' is up into an (X, Y) movement vector where 'Y' is forward.
+"""
+func _get_xy_velocity() -> Vector2:
+	return Vector2(_velocity.x, _velocity.z)
+
+
+"""
+Translates an (X, Y) movement vector where 'Y' is forward into Turbo's (X, Y, Z) velocity vector where 'Y' is up.
```

Comment:
Maybe `Assigns Turbo's (X, Y Z) velocity vector where 'Y' is up from an (X, Y) movement vector where 'Y' is forward`... Most importantly, using `Assigns` instead of `Translates` since translate has a specific meaning in 3D

Reply:
Changed to `Sets Turbo's (X, Y, Z) velocity...`